lib/active_record_shards/tasks.rb

```diff
@@ -31,11 +37,18 @@
 
   desc "Create the database defined in config/database.yml for the current RAILS_ENV including shards"
   task :create => :load_config do
-    env_name = defined?(Rails.env) ? Rails.env : RAILS_ENV || 'development'
     ActiveRecord::Base.configurations.each do |key, conf|
       if key.starts_with?(env_name) && !key.ends_with?("_slave")
         if ActiveRecord::VERSION::MAJOR >= 4
-          ActiveRecord::Tasks::DatabaseTasks.create(conf)
+          begin
+            root_connection(conf).create_database(conf['database'])
+          rescue ActiveRecord::StatementInvalid => ex
+            if ex.message.match('database exists')
+              $stderr.puts "#{conf['database']} already exists"
```

Comment:
Ah, I was doing what Rails itself does: https://github.com/rails/rails/blob/d26704a15f88d384dd282425daa832affdb5f8c1/activerecord/lib/active_record/tasks/database_tasks.rb#L95

But I can still change it if you want. I've no strong opinion on it.